src/providers/ad/ad_gpo.c

```diff
@@ -3601,9 +3608,48 @@ ad_gpo_populate_candidate_gpos(TALLOC_CTX *mem_ctx,
         goto done;
     }
 
+    i = som_count -1 ;
+    while (i >= 0) {
+        gp_som = som_list[i];
+
+        /* For unenforced_gpo_dns the most specific GPOs with the highest
+         * priority should be the last. We start with the top-level SOM and go
+         * down to the most specific one and add the unenforced following the
+         * gplink_list where the GPO with the highest priority comes last. */
+        j = 0;
+        while (gp_som && gp_som->gplink_list && gp_som->gplink_list[j]) {
+                gp_gplink = gp_som->gplink_list[j];
+                if (gp_gplink == NULL) {
```

Comment:
This check doesn't make sense due to `&& gp_som->gplink_list[j]` in a loop condition.

The same pattern is seen above (in original code).

(Btw, `gp_som &&` in inner loop doesn't make sense as well)